### PR TITLE
entc/gen: move eager-loading to method

### PR DESCRIPTION
**Pull request by a8m**
This is preparation work for `WithNamed<E>` API. For more context, see the proposal at https://github.com/ent/ent/issues/2755.

### CODE REVIEW

File: entc/gen/template/dialect/sql/query.tmpl [hunk 1]
```diff
@@ -88,6 +92,137 @@ func ({{ $receiver }} *{{ $builder }}) sqlAll(ctx context.Context, hooks ...quer
 	return nodes, nil
 }
 
+{{/* Generate a method to eager-load each edge. */}}
+{{- range $e := $.Edges }}
+	func ({{ $receiver }} *{{ $builder }}) load{{ $e.StructField }}(ctx context.Context, query *{{ $e.Type.QueryName }}, nodes []*{{ $.Name }}, init func(*{{ $.Name }}), assign func(*{{ $.Name }}, *{{ $e.Type.Name }})) error {
+		{{- if $e.M2M }}
+			edgeids := make([]driver.Value, len(nodes))
+			byid := make(map[{{ $.ID.Type }}]*{{ $.Name }})
```
Comment (masseelch) [inline entc/gen/template/dialect/sql/query.tmpl:100]:
```suggestion
			byids := make(map[{{ $.ID.Type }}]*{{ $.Name }})
```

Stay consistent

**Review by masseelch — COMMENTED** [review]
:100:

**Review by masseelch — APPROVED** [review]

File: entc/gen/template/dialect/sql/query.tmpl (hunk 1 above)
Comment (giautm) [reply entc/gen/template/dialect/sql/query.tmpl:100]: Should be `edgeIDs` and `byIDs`

**Review by giautm — APPROVED** [review]
💯

File: entc/gen/template/dialect/sql/query.tmpl (hunk 1 above)
Comment (a8m) [reply entc/gen/template/dialect/sql/query.tmpl:100]: I actually prefer `byid` as it describes the operation for getting a node by its ID and that's why this variable is used for.  I use this singular form quite often in ent and atlas. I'll keep it this way, but will change the naming to camelCase.

**Pull request merged**